Index: b/gtk/gtkplacessidebar.c
===================================================================
--- a/gtk/gtkplacessidebar.c
+++ b/gtk/gtkplacessidebar.c
@@ -176,6 +176,7 @@
   guint show_other_locations   : 1;
   guint show_trash             : 1;
   guint show_starred_location  : 1;
+  guint local_only             : 1;
 };
 
 struct _GtkPlacesSidebarClass {
@@ -231,6 +232,7 @@
   PROP_SHOW_ENTER_LOCATION,
   PROP_SHOW_TRASH,
   PROP_SHOW_STARRED_LOCATION,
+  PROP_LOCAL_ONLY,
   PROP_SHOW_OTHER_LOCATIONS,
   NUM_PROPERTIES
 };
@@ -633,6 +633,7 @@ open_home (GtkPlacesSidebar *sidebar)
 static void
 add_special_dirs (GtkPlacesSidebar *sidebar)
 {
+  return;
   GList *dirs;
   gint index;
 
@@ -1427,8 +1428,6 @@ update_places (GtkPlacesSidebar *sidebar
       root = sl->data;
       is_native = g_file_is_native (root);
 
-      if (_gtk_bookmarks_manager_get_is_builtin (sidebar->bookmarks_manager, root))
-        continue;
 
       if (sidebar->local_only && !is_native)
         continue;
@@ -1536,6 +1535,17 @@ update_places (GtkPlacesSidebar *sidebar
 
       g_object_unref (start_icon);
     }
+  else if (!sidebar->local_only)
+    {
+      start_icon = g_themed_icon_new_with_default_fallbacks (ICON_NAME_NETWORK);
+
+      add_place (sidebar, GTK_PLACES_BUILT_IN,
+                 PROP_SHOW_OTHER_LOCATIONS,
+                 _("Networks"), start_icon, NULL, "network:///",
+                 NULL, NULL, NULL, NULL, 0, _("Show other locations"));
+
+      g_object_unref (start_icon);
+    }
 
   gtk_widget_show_all (GTK_WIDGET (sidebar));
   /* We want this hidden by default, but need to do it after the show_all call */
@@ -4155,7 +4165,7 @@ gtk_places_sidebar_init (GtkPlacesSideba
 
   gtk_container_add (GTK_CONTAINER (sidebar), sidebar->list_box);
 
-  sidebar->hostname = g_strdup (_("Computer"));
+  sidebar->hostname = g_strdup (_("File System"));
   sidebar->hostnamed_cancellable = g_cancellable_new ();
   g_dbus_proxy_new_for_bus (G_BUS_TYPE_SYSTEM,
                             G_DBUS_PROXY_FLAGS_GET_INVALIDATED_PROPERTIES,
